In [ ]:
import pandas as pd
x=pd.read_csv(r"C:\Temp\Heart-Failure-Prediction\x.csv")
y=pd.read_csv(r"C:\Temp\Heart-Failure-Prediction\y.csv")
x_rus=pd.read_csv(r"C:\Temp\Heart-Failure-Prediction\x_rus.csv")
y_rus=pd.read_csv(r"C:\Temp\Heart-Failure-Prediction\y_rus.csv")
x_ros=pd.read_csv(r"C:\Temp\Heart-Failure-Prediction\x_ros.csv")
y_ros=pd.read_csv(r"C:\Temp\Heart-Failure-Prediction\y_ros.csv")
x_smt=pd.read_csv(r"C:\Temp\Heart-Failure-Prediction\x_smt.csv")
y_smt=pd.read_csv(r"C:\Temp\Heart-Failure-Prediction\y_smt.csv")
x_ada=pd.read_csv(r"C:\Temp\Heart-Failure-Prediction\x_ada.csv")
y_ada=pd.read_csv(r"C:\Temp\Heart-Failure-Prediction\y_ada.csv")

In [33]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test =train_test_split(x_ada,y_ada,test_size=0.2,random_state=42)

In [34]:
import xgboost as xgb
from sklearn.metrics import accuracy_score

xgb_model=xgb.XGBClassifier(objective='binary:logistic',eval_metric='logloss',random_state=42)
xgb_model.fit(x_train,y_train)
y_predict=xgb_model.predict(x_test)
accuracy=accuracy_score(y_predict,y_test)
print(f'accurecy is {accuracy}')

accurecy is 0.8780487804878049


In [44]:
from sklearn.model_selection import RandomizedSearchCV

param_dist = {
    'n_estimators': [100, 300, 500,800],
    'max_depth': [3, 5, 7, 10],
    'learning_rate': [0.01, 0.05, 0.1, 0.2],
    'subsample': [0.6, 0.8, 1.0],
    'gamma': [0, 1, 5],
    'reg_alpha': [0, 0.1, 1],
    'reg_lambda': [1, 5, 10]
}

random_search = RandomizedSearchCV(
    xgb.XGBClassifier(objective='binary:logistic', eval_metric='logloss', random_state=42),
    param_distributions=param_dist,
    n_iter=50,  
    scoring='accuracy',
    cv=10,
    verbose=1,
    n_jobs=-1
)

random_search.fit(x_train, y_train)
print("Best parameters:", random_search.best_params_)
print("Best accuracy:", random_search.best_score_)


Fitting 10 folds for each of 50 candidates, totalling 500 fits
Best parameters: {'subsample': 1.0, 'reg_lambda': 1, 'reg_alpha': 0, 'n_estimators': 500, 'max_depth': 7, 'learning_rate': 0.2, 'gamma': 0}
Best accuracy: 0.9053030303030305


Fitting 5 folds for each of 50 candidates, totalling 250 fits
Best parameters: {'subsample': 0.8, 'reg_lambda': 5, 'reg_alpha': 0, 'n_estimators': 100, 'max_depth': 10, 'learning_rate': 0.2, 'gamma': 0}
Best accuracy: 0.9020046620046619
---------------------
Fitting 10 folds for each of 50 candidates, totalling 500 fits
Best parameters: {'subsample': 0.8, 'reg_lambda': 10, 'reg_alpha': 0, 'n_estimators': 500, 'max_depth': 7, 'learning_rate': 0.05, 'gamma': 0}
Best accuracy: 0.9021780303030305
---------------------
Fitting 10 folds for each of 50 candidates, totalling 500 fits
Best parameters: {'subsample': 1.0, 'reg_lambda': 1, 'reg_alpha': 0, 'n_estimators': 500, 'max_depth': 7, 'learning_rate': 0.2, 'gamma': 0}
Best accuracy: 0.9053030303030305